backend/de.metas.business/src/main/java/de/metas/order/impl/OrderLineBL.java

```diff
@@ -888,4 +904,69 @@ public void save(final org.compiere.model.I_C_OrderLine orderLine)
 	{
 		orderDAO.save(orderLine);
 	}
+
+	@Override
+	@NonNull
+	public CurrencyPrecision extractPricePrecision(@NonNull final org.compiere.model.I_C_OrderLine olRecord)
+	{
+		return extractPriceList(olRecord)
+				.filter(priceListRecord -> !isNull(priceListRecord, I_M_PriceList.COLUMNNAME_PricePrecision))
+				.filter(priceListRecord -> priceListRecord.getPricePrecision() >= 0)
+				.map(I_M_PriceList::getPricePrecision)
+				.map(CurrencyPrecision::ofInt)
+				.orElseGet(() -> getPrecisionFromCurrency(olRecord));
+	}
+
+	@NonNull
+	private Optional<I_M_PriceList> extractPriceList(@NonNull final org.compiere.model.I_C_OrderLine olRecord)
+	{
+		final PriceListVersionId priceListVersionId = PriceListVersionId.ofRepoIdOrNull(olRecord.getM_PriceList_Version_ID());
+
+		if (priceListVersionId != null)
+		{
+			final I_M_PriceList result = priceListDAO.getPriceListByPriceListVersionId(priceListVersionId);
+			logger.debug("C_OrderLine {} has M_PriceList_Version_ID={}; -> return M_PriceList={}", olRecord.getC_OrderLine_ID(), priceListVersionId, result);
+			return Optional.ofNullable(result);
+		}
+
+		final PricingSystemId orderPricingSystemId = PricingSystemId.ofRepoIdOrNull(olRecord.getBase_PricingSystem_ID());
```

Comment:
do you know when the price list version ID is not set?

did a quick check on a big database and found ZERO cases:
| m\_pricelist\_version\_id is null | count |
| :--- | :--- |
| false | 464738 |
SQL:
```sql
SELECT m_pricelist_version_id IS NULL AS "m_pricelist_version_id is null"
     , COUNT(1)
FROM c_orderline
GROUP BY m_pricelist_version_id IS NULL
```


So, IMHO I would assume the price list version ID is set,
and I would skip checking in other places.

@metas-ts  wdyt?

PS: 
> PricingSystemId.ofRepoIdOrNull(olRecord.getBase_PricingSystem_ID());

it's not the order pricing system, that's something else.
the order pricing system is C_Order.M_PricingSystem_ID.

Reply:
> do you know when the price list version ID is not set?

nope, but it's not mandatory on the model so didn't want to risk it

> it's not the order pricing system, that's something else. the order pricing system is C_Order.M_PricingSystem_ID.

lol, right 🤦‍♂️